api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2022.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// AdminNetworkPolicy is the Schema for the adminnetworkpolicies API
+type AdminNetworkPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of AdminNetworkPolicy.
+	// +optional
+	Spec AdminNetworkPolicySpec `json:"spec,omitempty"`
+
+	// ANPStatus is the status to be reported by the implementation, this is not
+	// standardized in alpha and consumers should report what they see fit in
+	// relation to their AdminNetworkPolicy implementation
+	// +optional
+	Status AdminNetworkPolicyStatus `json:"status,omitempty"`
+}
+
+// AdminNetworkPolicyStatus defines the observed state of AdminNetworkPolicy
+type AdminNetworkPolicyStatus struct {
+	Conditions []metav1.Condition `json:"status"`
+}
+
+// AdminNetworkPolicySubject defines what objects the policy selects.
+// Exactly one of the `NamespaceSelector` or `NamespaceAndPodSelector` pointers
+// should be set.
+type AdminNetworkPolicySubject struct {
+	NamespaceSelector       *metav1.LabelSelector `json:"namespaceselector"`
+	NamespaceAndPodSelector *NamespacedPodSubject `json:"namespaceandpodselector"`
+}
+
+// NamespacedPodSubject allows the user to select a given set of pod(s) in
+// selected namespace(s)
+type NamespacedPodSubject struct {
+	// This field follows standard label selector semantics; if present but empty,
+	// it selects all Namespaces.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceselector"`
+
+	// Used to explicitly select pods within a namespace; if present but empty,
+	// it selects all Pods.
+	PodSelector *metav1.LabelSelector `json:"podselector"`
+}
+
+// AdminNetworkPolicySpec defines the desired state of AdminNetworkPolicy
+type AdminNetworkPolicySpec struct {
+	// Priority is an int32 value bound to 0 - 1000, the lowest positive priority,
+	// "1" corresponds to the highest importance, while higher priorities have
+	// lower importance. An ANP with a priority of "0" will be evaluated after all
+	// positive priority AdminNetworkPolicies and standard NetworkPolicies.
+	// This field is NOT optional.
+	Priority *int32 `json:"priority"`
```

Comment:
I think we'll probably want kubebuilder defaulting tags so we get defaulting performed on the CRD API.

Reply:
Add defaults where relevant